e2e-tests/adapters/gatsby-node.ts

```diff
@@ -62,3 +63,128 @@ export const createPages: GatsbyNode["createPages"] = ({
     context: {},
   })
 }
+
+// Image CDN
+export const createSchemaCustomization: GatsbyNode["createSchemaCustomization"] =
+  function createSchemaCustomization({ actions, schema, store }) {
+    actions.createTypes(`
+    type UnsplashImage implements Node & RemoteFile {
+      id: ID!
+    }
+
+    type DeployedLocalImage implements Node & RemoteFile {
+      id: ID!
+    }
+  `)
+
+    actions.createTypes(
+      addRemoteFilePolyfillInterface(
+        schema.buildObjectType({
+          name: "MyRemoteFile",
+          fields: {},
+          interfaces: ["Node", "RemoteFile"],
+        }),
+        {
+          schema,
+          actions,
+          store,
+        }
+      )
+    )
+  }
+
+export const sourceNodes: GatsbyNode["sourceNodes"] = function sourceNodes({
+  actions,
+  createNodeId,
+  createContentDigest,
+}) {
+  {
+    const imageURL = `https://images.unsplash.com/photo-1672823841196-3ec078a2befd`
+    actions.createNode({
+      id: "unsplash-image-1",
+      internal: {
+        type: "UnsplashImage",
+        contentDigest: `1`,
+      },
+      url: imageURL,
+      filename: imageURL,
+      mimeType: `image/jpeg`,
+      width: 1940,
+      height: 3118,
+    })
+  }
+
+  {
+    const imageURL = `${process.env.DEPLOY_URL}/images/cornwall.jpg`
+    actions.createNode({
+      id: "deployed-local-image-1",
+      internal: {
+        type: "DeployedLocalImage",
+        contentDigest: `1`,
+      },
+      url: imageURL,
+      filename: imageURL,
+      mimeType: `image/jpeg`,
+      width: 2400,
+      height: 1136,
+    })
+  }
+
+  {
+    const imageURL = `${process.env.DEPLOY_URL}/images/landsend.jpg`
+    actions.createNode({
+      id: "deployed-local-image-2",
+      internal: {
+        type: "DeployedLocalImage",
+        contentDigest: `2`,
+      },
+      url: imageURL,
+      filename: imageURL,
+      mimeType: `image/jpeg`,
+      width: 640,
+      height: 480,
+    })
+  }
+
```

Comment:
this should be removed in favor of MyRemoteFile (which is copied from e2e-tests/production)
```suggestion
```